jasminewd2/index.d.ts

```diff
@@ -44,6 +44,13 @@ declare namespace jasmine {
     toThrowError(expected?: new (...args: any[]) => Error | Promise<new (...args: any[]) => Error>, message?: string | RegExp | Promise<string | RegExp>): Promise<void>;
   }
 
+  interface ArrayLikeMatchers<T extends ArrayLike<T>> extends Matchers<T> {
```

Comment:
I am fine with `ArrayLike` over `Array` (as `Array`/`[]` __should be/implement__ `ArrayLike`).

I was confused about `T extends ArrayLike<T>`. From my understanding that looks like something that would be (could only be) an array of arrays.

For example, let's say you have something like `MyIntList`. That would satisfy `U extends ArrayLike<T>`: `MyIntList extends ArrayLike<number>`.